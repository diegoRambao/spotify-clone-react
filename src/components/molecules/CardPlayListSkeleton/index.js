import { Skeleton } from 'components/atoms/Skeleton'

export function CardPlaylistSkeleton () {
  return (
    <div>
      <Skeleton variant='rect' height='8rem' />
      <Skeleton variant='text' />
      <Skeleton variant='text' width='6rem' />
    </div>
  )
}
