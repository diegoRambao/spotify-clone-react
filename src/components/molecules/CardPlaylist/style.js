import styled from 'styled-components'
import { colors } from 'consts/styles'
import { Link as Navigate } from 'react-router-dom'

export const Link = styled(Navigate)`
    text-decoration: none;
    color: ${colors.white};
    padding: 14px;
    display: block;
    background-color: ${colors.darkGrayLight};
    width: auto;
    border-radius: 3px;
`

export const CardContent = styled.div`
    display: flex;
    align-items: center;
    gap: 1rem;
`

export const CardImg = styled.img`
    width: 50px;
    height: 50px;
    object-fit: contain;
    border-radius: 4px;
`

export const CardBody = styled.div``

export const CardTitle = styled.h4`
    margin: 0;
    font-size: 1rem;
    margin-bottom: 5px;
`

export const CardSubtitle = styled.h6`
    margin: 0;
    font-weight: 400;
    font-size: .7rem;
`
