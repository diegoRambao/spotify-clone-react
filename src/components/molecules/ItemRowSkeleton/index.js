import { Skeleton, Box } from 'components/atoms'

export function ItemRowSkeleton () {
  return (
    <Box className='d-flex justify-content-between'>
      <Box className='d-flex'>
        <Skeleton variant='rect' width='3rem' height='3rem' />
        <Box ml='.5rem' className='d-flex flex-column'>
          <Skeleton variant='text' width='4.5rem' height='.5rem' />
          <Skeleton variant='text' width='3rem' height='.5rem' />
        </Box>
      </Box>
      <Skeleton variant='text' width='3rem' />
      <Skeleton variant='text' width='3rem' />
    </Box>
  )
}
