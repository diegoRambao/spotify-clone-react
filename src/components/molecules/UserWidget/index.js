import styled from 'styled-components'
import { Avatar } from 'components/atoms/Avatar'
import { firstLeetterToUpperCase } from 'utils'

const Container = styled.div`
    display: flex;
    gap: 8px;
    align-items: center;
`

const Title = styled.span`
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    width: 200px;

    @media only screen and (min-width: 768px) {
        width: 130px;
    }
`

export function UserWidget ({ name = ' ', img }) {
  const firstLetter = firstLeetterToUpperCase(name)
  return (
    <Container>
      {img === null
        ? <Avatar>{firstLetter}</Avatar>
        : <Avatar src={img} alt={name} />}

      <Title>{name}</Title>
    </Container>
  )
}
