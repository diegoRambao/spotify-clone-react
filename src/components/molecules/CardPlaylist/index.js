import { Link, CardContent, CardImg, CardBody, CardTitle, CardSubtitle } from './style'
import playlistEmpty from 'assets/images/playlist-empty.png'

export function CardPlaylist ({ playlist }) {
  const { id, name, tracks, images } = playlist
  return (
    <Link to={`/playlist/${id}`} className='animate__fadeIn'>
      <CardContent>
        <CardImg
          loading='lazy'
          src={images[0]?.url || playlistEmpty}
          alt={name}
        />
        <CardBody>
          <CardTitle>{name}</CardTitle>
          <CardSubtitle>{tracks.total} canciones</CardSubtitle>
        </CardBody>
      </CardContent>
    </Link>
  )
}
