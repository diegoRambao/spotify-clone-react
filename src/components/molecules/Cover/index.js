import { Container, Img, Title, SubTitle } from './style'

export function Cover ({ img, title, subtitle }) {
  return (
    <Container>
      <Img src={img} alt={Title} />
      <div>
        <Title>{title}</Title>
        <SubTitle>{subtitle}</SubTitle>
      </div>
    </Container>
  )
}
