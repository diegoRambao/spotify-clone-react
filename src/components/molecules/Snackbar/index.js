// import { useState, useEffect } from 'react'
import { Container, Button, ContainerWith } from './style'
import { Box } from 'components/atoms'
import { CgClose } from 'react-icons/cg'
import { BsCheckCircle } from 'react-icons/bs'
import { BiErrorCircle } from 'react-icons/bi'
import { IoMdWarning } from 'react-icons/io'
import { AiOutlineInfoCircle } from 'react-icons/ai'

const iconsSnackSeverity = {
  error: <BiErrorCircle color='#fff' />,
  warning: <IoMdWarning color='#fff' />,
  info: <AiOutlineInfoCircle color='#fff' />,
  success: <BsCheckCircle color='#fff' />
}

export function Snackbar ({ open, onClose, children, severity }) {
//   console.log(open)
//   const [show, setShow] = useState(open)

  //   useEffect(() => {
  //     setShow(open)
  //   }, [open])

  return (
    open && (
      <ContainerWith>
        <Container severity={severity}>
          <Box className='d-flex align-items-center'>
            {severity && iconsSnackSeverity[severity]}
            <p>{children}</p>
          </Box>
          <Button onClick={onClose}>
            <CgClose />
          </Button>
        </Container>
      </ContainerWith>
    )
  )
}
