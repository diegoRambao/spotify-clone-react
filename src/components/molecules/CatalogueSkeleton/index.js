import { Box, Skeleton } from 'components/atoms'
import { ItemRowSkeleton } from '../ItemRowSkeleton'

export function CatalogueSkeleton () {
  return (
    <>
      <Box className='d-flex'>
        <Skeleton width='10rem' height='10rem' variant='rect' />
        <Box ml='0.5rem' className='d-flex flex-column justify-content-center'>
          <Skeleton width='7rem' variant='text' />
          <Skeleton width='5rem' height='.5rem' variant='text' />
        </Box>
      </Box>
      <Box mt='2rem'>
        <ItemRowSkeleton />
        <ItemRowSkeleton />
      </Box>
    </>
  )
}
