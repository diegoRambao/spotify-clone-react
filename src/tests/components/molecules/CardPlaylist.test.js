import { shallow, mount } from 'enzyme'
import { CardPlaylist } from 'components/molecules/CardPlaylist'
import { Link, MemoryRouter } from 'react-router-dom'

describe('<CardPlaylist>', () => {
  const playlist = {
    id: '1234567890', name: 'Playlist 1', tracks: { total: 67 }, images: [{ url: 'https://example.png' }]
  }

  const wrapper = mount(<MemoryRouter><CardPlaylist playlist={playlist} /></MemoryRouter>)

  test('render content', () => {
    const wrapper = shallow(<CardPlaylist playlist={playlist} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('property to link to be id playlist', () => {
    const link = wrapper.find(Link)
    expect(link.prop('to')).toBe(`/playlist/${playlist.id}`)
  })

  test('render image', () => {
    const img = wrapper.find('img')
    expect(img.prop('alt')).toBe(playlist.name)
    expect(img.prop('src')).toBe(playlist.images[0].url)
    expect(img.prop('loading')).toBe('lazy')
  })

  test('render title', () => {
    const h4 = wrapper.find('h4')
    expect(h4.text()).toBe(playlist.name)
  })

  test('render total', () => {
    const h6 = wrapper.find('h6')
    expect(h6.text()).toBe(`${playlist.tracks.total} canciones`)
  })
})
