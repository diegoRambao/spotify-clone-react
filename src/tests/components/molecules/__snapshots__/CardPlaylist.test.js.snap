// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardPlaylist> render content 1`] = `
<Styled(Link)
  className="animate__fadeIn"
  to="/playlist/1234567890"
>
  <styled.div>
    <styled.img
      alt="Playlist 1"
      loading="lazy"
      src="https://example.png"
    />
    <styled.div>
      <styled.h4>
        Playlist 1
      </styled.h4>
      <styled.h6>
        67
         canciones
      </styled.h6>
    </styled.div>
  </styled.div>
</Styled(Link)>
`;
